src/OrleansAzureUtils/Storage/RemindersTableManager.cs

```diff
@@ -118,55 +118,57 @@ private RemindersTableManager(Guid serviceId, string deploymentId, string storag
-                        && String.Compare(e.PartitionKey, sBegin) > 0
-                        && String.Compare(e.PartitionKey, sEnd) <= 0;
-
-                var queryResults = await ReadTableEntriesAndEtagsAsync(query);
+                string filterBetweenBeginAndEnd = TableQuery.CombineFilters(
+                    TableQuery.GenerateFilterCondition("PartitionKey", QueryComparisons.GreaterThan, sBegin),
+                    TableOperators.And,
+                    TableQuery.GenerateFilterCondition("PartitionKey", QueryComparisons.LessThanOrEqual,
+                        sEnd));
+                string query = TableQuery.CombineFilters(filterOnServiceIdStr, TableOperators.And, filterBetweenBeginAndEnd);
+                var queryResults = await ReadTableEntriesAndEtagsWithFilterStringAsync(query);
                 return queryResults.ToList();
             }
 
             if (begin == end)
             {
-                
-                Expression<Func<ReminderTableEntry, bool>> query =
-                    e => String.Compare(e.PartitionKey, serviceIdStr + '_') > 0
-                        && String.Compare(e.PartitionKey, serviceIdStr + (char)('_' + 1)) <= 0;
-
-                var queryResults = await ReadTableEntriesAndEtagsAsync(query);
+                var queryResults = await ReadTableEntriesAndEtagsWithFilterStringAsync(filterOnServiceIdStr);
                 return queryResults.ToList();
             }
 
             // (begin > end)
-            Expression<Func<ReminderTableEntry, bool>> p1Query =
-                e => String.Compare(e.PartitionKey, serviceIdStr + '_') > 0
-                    && String.Compare(e.PartitionKey, serviceIdStr + (char)('_' + 1)) <= 0
-                    && String.Compare(e.PartitionKey, sBegin) > 0;
-            Expression<Func<ReminderTableEntry, bool>> p2Query =
-                e => String.Compare(e.PartitionKey, serviceIdStr + '_') > 0
-                    && String.Compare(e.PartitionKey, serviceIdStr + (char)('_' + 1)) <= 0
-                    && String.Compare(e.PartitionKey, sEnd) <= 0;
-
-            var p1 = ReadTableEntriesAndEtagsAsync(p1Query);
-            var p2 = ReadTableEntriesAndEtagsAsync(p2Query);
-
-            IEnumerable<Tuple<ReminderTableEntry, string>>[] arr = await Task.WhenAll(p1, p2);
-            return  arr[0].Concat(arr[1]).ToList();
+            string filterExtendBeginAndEnd = TableQuery.CombineFilters(
+                    TableQuery.GenerateFilterCondition("PartitionKey", QueryComparisons.GreaterThan, sBegin),
+                    TableOperators.Or,
+                    TableQuery.GenerateFilterCondition("PartitionKey", QueryComparisons.LessThanOrEqual,
+                        sEnd));
+            string query1 = TableQuery.CombineFilters(
+                filterOnServiceIdStr, TableOperators.And, filterExtendBeginAndEnd);
+
+            var results = await ReadTableEntriesAndEtagsWithFilterStringAsync(query1);
+            return results.ToList();
```

Comment:
Why definitely? Just asking, because at least at hindsight it doesn't look like the new approach should be better than the old one. Still, sure, let's discuss in a different PR, but let's not change the behavior implicitly when porting